pyproject.toml

```diff
@@ -40,12 +40,20 @@ dependencies = [
 dev = [
     "pytest",
     "pytest-asyncio",
-    "nbval",            # For notebook testing
+    "nbval",                      # For notebook testing
     "black",
     "ruff",
     "types-requests",
     "types-setuptools",
     "pre-commit",
+    "pytest-html",
```

Comment:
so this set of dependencies depends on which distro or provider set you are running for. adding all the deps for "dev" is going to make some folks unhappy -- especially the large "torch" dependency which will bring in cuda.

for now, how about we create a ".[test]" sub-package which is focused on running tests? for this package, you can add the CPU dependencies for torch. See 
<img width="808" alt="image" src="https://github.com/user-attachments/assets/ef8f679f-e88d-4747-b3af-3dc44dd361c9" />

Reply:
Agreed. This needs to be optional optional deps